src/main/resources/com/google/api/codegen/csharp/gapic_client.snip

```diff
@@ -0,0 +1,465 @@
+        
+        @join method : api.apiMethodsImpl
+            {@xmlDoc(method.doc)}
+            {@method(method, TRUE)}
+
+        @end
+    }
+@end
+
+@private method(method, isImpl)
+    @switch method.type.toString
+    @case "FlattenedMethod"
+        {@flattenedMethodSync(method, isImpl)}
+    @case "FlattenedAsyncCallSettingsMethod"
+        {@flattenedMethodAsyncCallSettings(method, isImpl)}
+    @case "FlattenedAsyncCancellationTokenMethod"
+        {@flattenedMethodAsyncCancellationToken(method, isImpl)}
+    @case "PagedFlattenedMethod"
+        {@pagedFlattenedMethodSync(method, isImpl)}
+    @case "PagedFlattenedAsyncMethod" 
+        {@pagedFlattenedMethodAsync(method, isImpl)}
+    @default
+        *** ERROR: Cannot handle method type '{@method.type.toString}' ***
+    @end
+@end
+
+@private flattenedMethodSync(method, isImpl)
+    @if isImpl
+        public override {@method.responseTypeName} {@method.name}(
+        {@""}    {@objectParams(method.methodParams)}) => {@method.callableName}.Sync(
+        {@""}        new {@method.apiRequestTypeName}
+        {@""}        {
+                    {@requestProperties(method.forwardingMethodParams)}
+        {@""}        },
+        {@""}        callSettings);
+    @else
+        public virtual {@method.responseTypeName} {@method.name}(
+        {@""}    {@objectParams(method.methodParams)})
+        {
+            throw new NotImplementedException();
+        }
+    @end
+@end
+
+@private flattenedMethodAsyncCallSettings(method, isImpl)
+    @if isImpl
+        public override {@method.responseTypeName} {@method.name}(
+        {@""}    {@objectParams(method.methodParams)}) => {@method.callableName}.Async(
+        {@""}        new {@method.apiRequestTypeName}
+        {@""}        {
+                    {@requestProperties(method.forwardingMethodParams)}
+        {@""}        },
+        {@""}        callSettings);
+    @else
+        public virtual {@method.responseTypeName} {@method.name}(
+        {@""}    {@objectParams(method.methodParams)})
+        {
+            throw new NotImplementedException();
+        }
+    @end
+@end
+
+@private flattenedMethodAsyncCancellationToken(method, isImpl)
+    @if isImpl
+        *** ERROR: Should never get here ***
+    @else
+        public virtual {@method.responseTypeName} {@method.name}(
+            {@objectParams(method.methodParams)}) => {@method.name}(
+                {@objectArgs(method.forwardingMethodParams)},
+                new CallSettings { CancellationToken = cancellationToken });
+    @end
+@end
+
+@private pagedFlattenedMethodSync(method, isImpl)
+    @if isImpl
+        public override IPagedEnumerable<{@method.pagedView.responseTypeName}, {@method.pagedView.resourceTypeName}> {@method.name}(
+        {@""}    {@objectParams(method.methodParams)}) => new PagedEnumerable<{@method.pagedView.requestTypeName}, {@method.pagedView.responseTypeName}, {@method.pagedView.resourceTypeName}>(
```

Comment:
You shouldn't need the `{@""}` here. The only time you need it in a place like this is when you want to force an indentation on the first line of a snippet that has no raw text lines before it.